yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -955,22 +955,27 @@ class ArchiverStoreHelper
     // from - blocksToUnwind = the new head, so + 1 for what we need to remove
     const blocks = await this.getBlocks(from - blocksToUnwind + 1, blocksToUnwind);
 
-    return [
+    const opResults = await Promise.all([
       // Unroll all logs emitted during the retrieved blocks and extract any contract classes and instances from them
-      ...(await Promise.all(
-        blocks.map(async block => {
-          const contractClassLogs = block.data.body.txEffects
-            .flatMap(txEffect => (txEffect ? [txEffect.contractClassLogs] : []))
-            .flatMap(txLog => txLog.unrollLogs());
-          // ContractInstanceDeployed event logs are broadcast in privateLogs.
-          const privateLogs = block.data.body.txEffects.flatMap(txEffect => txEffect.privateLogs);
-          await this.#updateRegisteredContractClasses(contractClassLogs, block.data.number, Operation.Delete);
-          await this.#updateDeployedContractInstances(privateLogs, block.data.number, Operation.Delete);
-        }),
-      )),
+      ...blocks.map(async block => {
+        const contractClassLogs = block.data.body.txEffects
+          .flatMap(txEffect => (txEffect ? [txEffect.contractClassLogs] : []))
+          .flatMap(txLog => txLog.unrollLogs());
+
+        // ContractInstanceDeployed event logs are broadcast in privateLogs.
+        const privateLogs = block.data.body.txEffects.flatMap(txEffect => txEffect.privateLogs);
+
+        return (
+          (await this.#updateRegisteredContractClasses(contractClassLogs, block.data.number, Operation.Delete)) &&
+          (await this.#updateDeployedContractInstances(privateLogs, block.data.number, Operation.Delete))
+        );
+      }),
+
       this.store.deleteLogs(blocks.map(b => b.data)),
       this.store.unwindBlocks(from, blocksToUnwind),
-    ].every(Boolean);
+    ]);
+
+    return opResults.every(Boolean);
```

Comment:
Change looks bigger than it actually is: I moved the `Promise.all` to the outside of the array rather than the inside.